core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -2379,9 +2379,9 @@ private static SqlNode createSqlDataTypeSpecByName(String typeAlias, RelDataType
         if (isContainsPrecision) {
           String dataType = getDataTypeBasedOnPrecision(precision, scale);
           if (!isContainsNegativePrecisionOrScale) {
-            typeAlias =
-                precision > 0 ? isContainsScale ? dataType + "(" + precision + ","
-                    + scale + ")" : dataType + "(" + precision + ")" : dataType;
+            typeAlias = precision > 0 && !(scale > 38) ? isContainsScale
```

Comment:
Here we have used value of scale before checking isContains scale, Please check if this is the best way to use both  .